cmd/oci-image-tool/autodetect.go

```diff
@@ -43,7 +42,7 @@ func autodetect(path string) (string, error) {
 	}
 
 	if fi.IsDir() {
-		return typeImageLayout, nil
```

Comment:
On Mon, Aug 29, 2016 at 06:52:32PM -0700, Stephen Day wrote:
> > @@ -43,7 +42,7 @@ func autodetect(path string) (string, error) {
> >  	}
> >  
> >  	if fi.IsDir() {
> > -		return typeImageLayout, nil
> 
> This doesn't seem right to just remove. Was this not correct before?

The new code handles the distinction between typeImageLayout and
typeImage (which is whether the target is a directory or tarball)
inside the NewEngine constructors [1,2].  I haven't added support for
the directory-based engines yet, since there was already enough going
on.  But I'm happy to add them if it would help with review.

[1]: https://github.com/opencontainers/image-spec/pull/159/files/1976fd6aaf0ef41527bf0434e7644310cc553a7d#diff-c3a5563e81df7aa2a94d70642ba7c3ceR28
[2]: https://github.com/opencontainers/image-spec/pull/159/files/1976fd6aaf0ef41527bf0434e7644310cc553a7d#diff-65bd80b0d4865c33446593b6c8e353f3R28